demo/demo/tests/test_apple_webhook.py

```diff
@@ -0,0 +1,264 @@
+                           original_transaction_id: str = 'test-original-transaction-id') -> AppStoreNotification:
+    result = AppStoreNotification.parse_obj(
+        {
+            'notification': AppStoreNotificationTypeV2.DID_RENEW.value if is_renew else AppStoreNotificationTypeV2.TEST.value,
+            'subtype': AppStoreNotificationTypeV2Subtype.ACCEPTED.value,
+            'notificationUUID': '00000000-0000-0000-0000-000000000000',
+            'data': {
+                'appAppleId': 12345,
+                'bundleId': bundle_id,
+                'bundleVersion': 'test-bundle-version',
+                'environment': AppleEnvironment.PRODUCTION.value,
+                'signedTransactionInfo': 'fake-transaction-info',
+            },
+        }
+    )
+
+    result.transaction_info.app_account_token = 'test-app-account-token'
+    result.transaction_info.bundle_id = bundle_id
+    result.transaction_info.purchase_date = datetime.datetime(2022, 4, 15, tzinfo=datetime.timezone.utc)
+    result.transaction_info.expires_date = datetime.datetime(2022, 5, 15, tzinfo=datetime.timezone.utc)
+    result.transaction_info.product_id = product_id
+    result.transaction_info.transaction_id = 'test-transaction-id'
+    result.transaction_info.original_transaction_id = original_transaction_id
+
+    return result
+
+
+def make_notification_query() -> dict:
+    return {'signedPayload': 'test-signed-payload'}
+
+
+def test__valid_receipt_sent(user_client, apple_in_app, apple_product_id, apple_bundle_id, user):
+    receipt_data = make_receipt_data(apple_product_id, apple_bundle_id)
+    with mock.patch(RECEIPT_FETCH_FUNCTION) as mock_receipt:
+        mock_receipt.return_value = receipt_data
+        response = user_client.post(APPLE_API_WEBHOOK, make_receipt_query(), content_type='application/json')
+
+    assert response.status_code == 200
+
+    payment = one(SubscriptionPayment.objects.all())
+    single_in_app = one(receipt_data.receipt.in_apps)
+    assert payment.user == user
+    assert payment.plan.metadata[apple_in_app.codename] == single_in_app.product_id
+    assert payment.status == SubscriptionPayment.Status.COMPLETED
+    assert payment.provider_codename == apple_in_app.codename
+    assert payment.provider_transaction_id == single_in_app.transaction_id
+    assert payment.subscription_start == single_in_app.purchase_date
+    assert payment.subscription_end == single_in_app.expires_date
+
+
+def test__invalid_receipt_sent(user_client, apple_in_app, apple_product_id, apple_bundle_id):
+    receipt_data = make_receipt_data(apple_product_id, apple_bundle_id, is_valid=False)
+    with mock.patch(RECEIPT_FETCH_FUNCTION) as mock_receipt:
+        mock_receipt.return_value = receipt_data
+        with pytest.raises(AppleReceiptValidationError):
+            user_client.post(APPLE_API_WEBHOOK, make_receipt_query(), content_type='application/json')
+
+    assert len(SubscriptionPayment.objects.all()) == 0
```

Comment:
`assert SubscriptionPayment.objects.count() == 0` or better `assert not SubscriptionPayment.objects.exists()`